util/chplenv/chpl_bin_subdir.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python
+
+import optparse
+import os
+from sys import stderr, stdout
+import sys
+
+chplenv_dir = os.path.dirname(__file__)
+sys.path.insert(0, os.path.abspath(chplenv_dir))
+
+import chpl_home_utils
+import chpl_platform, chpl_compiler, chpl_machine, chpl_arch
+
+from utils import memoize
+
+@memoize
+def get(location):
+    platform = chpl_platform.get(location)
+    compiler = chpl_compiler.get(location)
+    machine = chpl_machine.get(location)
+    (flag, arch) = chpl_arch.get(location, map_to_compiler=True,
+                                 get_lcd=chpl_home_utils.using_chapel_module())
+
+    # platform
+    result = platform
+    # compiler -- included for target
+    if location == 'target':
+      result += '-' + compiler
+    # machine
+    result += '-' + machine
+    # arch
+    if arch != 'none' and arch != 'unknown':
```

Comment:
Why not include arch here?

Reply:
It's my response to your suggestion:
https://github.com/chapel-lang/chapel/pull/11638#issuecomment-438765716

if the arch is none or unknown, we elide it from the bin/ path. Note that we're not planning to specialize the compiler build, so we need some rule to keep the bin/ subdirectory simple.